server/mqtt.go

```diff
@@ -1037,49 +1039,88 @@ func (s *Server) mqttCreateAccountSessionManager(acc *Account, quitCh chan struc
 		as.sessPersistProcessing(closeCh)
 	})
 
-	// Create the stream for the sessions.
-	cfg := &StreamConfig{
-		Name:       mqttSessStreamName,
-		Subjects:   []string{mqttSessStreamSubjectPrefix + as.domainTk + ">"},
-		Storage:    FileStorage,
-		Retention:  LimitsPolicy,
-		Replicas:   as.replicas,
-		MaxMsgsPer: 1,
-	}
-	if _, created, err := jsa.createStream(cfg); err == nil && created {
-		as.transferUniqueSessStreamsToMuxed(s)
-	} else if isErrorOtherThan(err, JSStreamNameExistErr) {
-		return nil, fmt.Errorf("create sessions stream for account %q: %v", acc.GetName(), err)
-	}
-
-	// Create the stream for the messages.
-	cfg = &StreamConfig{
-		Name:      mqttStreamName,
-		Subjects:  []string{mqttStreamSubjectPrefix + ">"},
-		Storage:   FileStorage,
-		Retention: InterestPolicy,
-		Replicas:  as.replicas,
-	}
-	if _, _, err := jsa.createStream(cfg); isErrorOtherThan(err, JSStreamNameExistErr) {
-		return nil, fmt.Errorf("create messages stream for account %q: %v", acc.GetName(), err)
-	}
-
-	// Create the stream for retained messages.
-	cfg = &StreamConfig{
-		Name:      mqttRetainedMsgsStreamName,
-		Subjects:  []string{mqttRetainedMsgsStreamSubject},
-		Storage:   FileStorage,
-		Retention: LimitsPolicy,
-		Replicas:  as.replicas,
-	}
-	si, _, err := jsa.createStream(cfg)
-	if isErrorOtherThan(err, JSStreamNameExistErr) {
-		return nil, fmt.Errorf("create retained messages stream for account %q: %v", acc.GetName(), err)
+	lookupStream := func(stream, txt string) (*StreamInfo, error) {
+		si, err := jsa.lookupStream(stream)
+		if err != nil {
+			if IsNatsErr(err, JSStreamNotFoundErr) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("lookup %s stream for account %q: %v", txt, accName, err)
+		}
+		if opts.MQTT.StreamReplicas == 0 {
+			return si, nil
+		}
+		sr := 1
+		if si.Cluster != nil {
+			sr += len(si.Cluster.Replicas)
+		}
+		if replicas != sr {
+			s.Warnf("account %q: %s stream replicas is %v, which does not match configured value of %v",
```

Comment:
Maybe use syntax we have elsewhere acc -> stream IIRC.
Also make sure log says MQTT since this is MQTT specific.

Reply:
Done